src/Thinktecture.Blazor.ScreenWakeLock/wwwroot/Thinktecture.Blazor.ScreenWakeLock.js

```diff
@@ -0,0 +1,70 @@
+export function isSupported() {
+    return isSupportedInWindow() || isSupportedInNavigator();
+}
+
+let wakeLock = null;
+
+export async function requestWakeLock(component, method) {
+    if (isSupportedInWindow()) {
+        requestWakeLockWindow(component, method);
+    } else if (isSupportedInNavigator()) {
+        await requestWakeLockNavigator(component, method);
+    }
+}
+
+export function releaseWakeLock() {
+    if (wakeLock) {
+        wakeLock.release();
+        wakeLock = null;
+    }
+}
+
+async function requestWakeLockNavigator(component, method) {
+    const requestWakeLock = async () => {
+        try {
+            wakeLock = await navigator.wakeLock.request('screen');
+            wakeLock.addEventListener('release', (e) => {
+                console.log(e);
+                console.log('Wake Lock was released');
+                component.invokeMethodAsync(method);
+            });
+            console.log('Wake Lock is active');
+        } catch (e) {
+            console.error(`${e.name}, ${e.message}`);
+            throw e;
+        }
+    };
+    
+    releaseWakeLock();
+    await requestWakeLock();
+}
+
+function requestWakeLockWindow(component, method) {
+    const requestWakeLock = () => {
+        const controller = new AbortController();
+        const signal = controller.signal;
+        window.WakeLock.request('screen', {signal})
```

Comment:
I don't know if we need to support this old version of the API. I think it would be fine to remove it.